report.md

```diff
@@ -35,7 +35,27 @@ We will use GraphQL to facilitate communication between the Client & the Server.
 1. SQL injections & XSS (especially if the input field is a custom type, such as JSON). Therefore, we will need to sanitise user input.
 1. Broken Access Controls. GraphQL does not verify whether a user has the permissions to retrieve sensitive data such as the password of the user, etc. Therefore, we will be using Apache Shiro to perform user permissions authentication.
 
-We will also be protecting our system by:
+Apache Shiro will be used to perform user permissions authentication in the following steps:
+1.	Collect the subject’s principals and credentials
+2.	Submit the principals and credentials to an authentication system.
+3.	Allow access, retry authentication, or block access
+
+The authentication system is represented in Shiro by security-specific [Realms](https://shiro.apache.org/realm.html). A Realm is a component that can access security data such as users, roles, and permissions. The Realm translates these data into a format that Shiro understands so Shiro can in turn provide a single easy-to-understand API.

```

Comment:
> The Realm translates these data into a format that Shiro understands so Shiro can in turn provide a single easy-to-understand API.

This sentence doesn't seem to be that relevant to the purpose of the design report; it seems to be just additional information.